src/repositories/soft-crud.repository.base.ts

```diff
@@ -194,13 +189,9 @@ export abstract class SoftCrudRepository<
     filter?: Filter<T>,
     options?: Options,
   ): Promise<T & Relations> {
-    //As parent method findById have filter: FilterExcludingWhere<T>
-    //so we need add check here.
-    const entityToRemove = await super.findOne(filter, options);
-
-    if (entityToRemove) {
-      // Now call super
-      return super.findById(id, filter, options);
+    const entity = await super.findById(id, filter, options);
+    if (entity) {
```

Comment:
```suggestion
 if (entity && !entity.deleted) {
```

Reply:
here we are returning even the soft deleted entity hence not added.
is that fine?

Reply:
got it